src/Notifier.cpp

```diff
@@ -163,9 +183,9 @@ size_t Notifier::ReceiveConfirmation() {
  * Reset the Shotgun-Alien queue to nullptr
  */
 void Notifier::Reset() {
-   std::lock_guard<std::mutex> guard(gLock);
    gQueue.reset(nullptr);
    gHandshakeQueue.reset(nullptr);
+   mKeepRunning.store(false);
```

Comment:
OK. If this is only used in a single threaded context then I assume that the destructor is still triggered in another thread, right?